script.rb

```diff
@@ -0,0 +1,93 @@
+module Enumerable
+  def my_each
+    return to_enum(:each) unless block_given?
+
+    arr = self if instance_of?(Array)
+    arr.length.times { |item| yield(arr[item]) }
+    self
+  end
+
+  def my_each_with_index
+    return to_enum(:each_with_index) unless block_given?
+
+    arr = self if instance_of?(Array)
+    arr.length.times { |item| yield(arr[item], item) }
+    self
+  end
+
+  def my_select
+    new_arr = []
+    return to_enum(:select) unless block_given?
+
+    my_each { |m| new_arr << m if yield(m) }
+    new_arr
+  end
+
+  def my_all?
+    condition = true
+    my_each do |item|
+      condition = false if (block_given? && !yield(item)) || (!block_given? && !item)
+    end
+    condition
+  end
+
+  def my_any?
+    condition = false
+    my_each do |item|
+      condition = true if (block_given? && yield(item)) || (!block_given? && item)
+    end
+    condition
+  end
+
+  def my_none?
+    condition = true
+    my_each do |item|
+      condition = false if (block_given? && yield(item)) || (!block_given? && item)
+    end
+    condition
+  end
+
+  def my_count
+    return size unless block_given?
+
+    count = 0
+    my_each do |item|
+      count += 1 if yield(item)
+    end
+    count
+  end
+
+  def my_map(input)
+    arr = to_a
+    new_map = []
+    arr.my_each do |item|
+      new_map << if block_given?
+                   yield(input.call(item))
+                 else
+                   input.call(item)
+                 end
+    end
+    new_map
+  end
```

Comment:
#my_map returns a new array with the results of running a given block once for every element in enum.
#my_map returns a new array with the results of running a given block once for every element in enum.
#my_map returns a new array with the results of running a given block once for every element in enum::range
#my_map returns an Enumerator if no block is given
#my_map accepts a proc as an argument and returns a new array with the results of running the 
proc once for every element in enum.
#my_map executes only the proc when both a block and a proc are given and returns a new array with the results of running the proc once for every element in enum.
#my_map does not mutate the original array